### ISSUE TITLE
gRPC interop: `large_unary` hangs

### ISSUE THREAD

**Issue by stevej** [linkerd/linkerd#929]
```
/Users/stevej/grpc-interop-testing-1.1.0-SNAPSHOT/bin% ./test-client --use_tls=false --server_port=60001 --test_case=large_unary
Jan 13, 2017 11:52:25 AM io.grpc.internal.ManagedChannelImpl <init>
INFO: [io.grpc.internal.ManagedChannelImpl-1] Created with target directaddress:///localhost/127.0.0.1:60001
Running test large_unary
^CShutting down
```

Logs from the server
```
[info] Running grpc.testing.ServerMain
Jan 13, 2017 11:52:22 AM com.twitter.finagle.Init$$anonfun$4 apply$mcV$sp
INFO: Finagle version 6.41.0 (rev=95eedf5f41f78414fae25d93cc8fae02eeb5a75d) built at 20161220-164342
Jan 13, 2017 11:52:26 AM io.netty.handler.codec.http2.Http2FrameLogger log
INFO:
----------------OUTBOUND--------------------
[id: 0xe2b74110, L:/127.0.0.1:60001 - R:/127.0.0.1:62833] SETTINGS: ack=false, settings={}
------------------------------------
Jan 13, 2017 11:52:26 AM io.netty.handler.codec.http2.Http2FrameLogger log
INFO:
----------------INBOUND--------------------
[id: 0xe2b74110, L:/127.0.0.1:60001 - R:/127.0.0.1:62833] SETTINGS: ack=false, settings={ENABLE_PUSH=0, MAX_CONCURRENT_STREAMS=0, INITIAL_WINDOW_SIZE=66560}
------------------------------------
Jan 13, 2017 11:52:26 AM io.netty.handler.codec.http2.Http2FrameLogger log
INFO:
----------------OUTBOUND--------------------
[id: 0xe2b74110, L:/127.0.0.1:60001 - R:/127.0.0.1:62833] SETTINGS: ack=true
------------------------------------
Jan 13, 2017 11:52:26 AM io.netty.handler.codec.http2.Http2FrameLogger log
INFO:
----------------INBOUND--------------------
[id: 0xe2b74110, L:/127.0.0.1:60001 - R:/127.0.0.1:62833] SETTINGS: ack=true
------------------------------------
Jan 13, 2017 11:52:26 AM io.netty.handler.codec.http2.Http2FrameLogger log
INFO:
----------------INBOUND--------------------
[id: 0xe2b74110, L:/127.0.0.1:60001 - R:/127.0.0.1:62833] HEADERS: streamId=3, headers=DefaultHttp2Headers[:authority: localhost:60001, :path: /grpc.testing.TestService/UnaryCall, :method: POST, :scheme: http, content-type: application/grpc, te: trailers, user-agent: grpc-java-netty/1.1.0-SNAPSHOT, grpc-accept-encoding: gzip, grpc-census-bin: ], streamDependency=0, weight=16, exclusive=false, padding=0, endStream=false
------------------------------------
Jan 13, 2017 11:52:26 AM io.netty.handler.codec.http2.Http2FrameLogger log
INFO:
----------------INBOUND--------------------
[id: 0xe2b74110, L:/127.0.0.1:60001 - R:/127.0.0.1:62833] DATA: streamId=3, padding=0, endStream=false, length=16384, bytes=00000425e010af96131ad8cb1012d4cb100000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000...
------------------------------------
Jan 13, 2017 11:52:26 AM io.netty.handler.codec.http2.Http2FrameLogger log
INFO:
----------------INBOUND--------------------
[id: 0xe2b74110, L:/127.0.0.1:60001 - R:/127.0.0.1:62833] DATA: streamId=3, padding=0, endStream=false, length=16384, bytes=00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000...
------------------------------------
Jan 13, 2017 11:52:26 AM io.netty.handler.codec.http2.Http2FrameLogger log
INFO:
----------------INBOUND--------------------
[id: 0xe2b74110, L:/127.0.0.1:60001 - R:/127.0.0.1:62833] DATA: streamId=3, padding=0, endStream=false, length=16384, bytes=00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000...
------------------------------------
Jan 13, 2017 11:52:26 AM io.netty.handler.codec.http2.Http2FrameLogger log
INFO:
----------------INBOUND--------------------
[id: 0xe2b74110, L:/127.0.0.1:60001 - R:/127.0.0.1:62833] DATA: streamId=3, padding=0, endStream=false, length=16244, bytes=00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000...
------------------------------------
Jan 13, 2017 11:52:26 AM io.netty.handler.codec.http2.Http2FrameLogger log
INFO:
----------------INBOUND--------------------
[id: 0xe2b74110, L:/127.0.0.1:60001 - R:/127.0.0.1:62833] DATA: streamId=3, padding=0, endStream=false, length=139, bytes=00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000...
------------------------------------
```
it then hangs until I reset the client and the following log items are output
```
Jan 13, 2017 11:52:55 AM com.twitter.finagle.buoyant.h2.netty4.Netty4DispatcherBase$$anonfun$loop$lzycompute$1$1 apply
ERROR: [S L:/127.0.0.1:60001 R:/127.0.0.1:62833] dispatcher failed
com.twitter.finagle.ChannelClosedException: ChannelException at remote address: /127.0.0.1:62833. Remote Info: Not Available

Jan 13, 2017 11:52:55 AM com.twitter.finagle.util.DefaultMonitor logWithRemoteInfo
WARNING: Exception propagated to the default monitor (upstream address: /127.0.0.1:62833, downstream address: n/a, label: ).
Reset.Cancel

Jan 13, 2017 11:52:55 AM io.netty.handler.codec.http2.Http2FrameLogger log
INFO:
----------------OUTBOUND--------------------
[id: 0xe2b74110, L:/127.0.0.1:60001 ! R:/127.0.0.1:62833] WINDOW_UPDATE: streamId=0, windowSizeIncrement=65535
------------------------------------
Jan 13, 2017 11:52:55 AM io.netty.handler.codec.http2.Http2FrameLogger log
INFO:
----------------OUTBOUND--------------------
[id: 0xe2b74110, L:/127.0.0.1:60001 ! R:/127.0.0.1:62833] WINDOW_UPDATE: streamId=3, windowSizeIncrement=65535
------------------------------------
```

**Comment by olix0r** [comment]
I'm not sure how this test is supposed to work:

from `test_util.go`:
```golang
	largeReqSize        = 271828
```

but with linkerd advertising a window size of 64K, we can never fit this message in the window.

We'll need to be able to configure linkerd with a large initial window size for this test to work.

**Comment by stevej** [comment]
I've noticed both grpc-go and grpc-java set monster window sizes.

public static final int DEFAULT_FLOW_CONTROL_WINDOW = 1048576; // 1MiB

https://github.com/grpc/grpc-java/blob/65e4d9f47abdad2a0a7a67fc5f1002bcf6f6fab1/netty/src/main/java/io/grpc/netty/NettyChannelBuilder.java#L69

**Comment by olix0r** [comment]
that's wild

**Comment by olix0r** [comment]
fixed in #1024 #1032
